Makefile

```diff
@@ -103,8 +110,16 @@ typedoc:
 docs: typedoc generated-documentation
 
 .PHONY: view-docs
-view-docs: typedoc
-	open ${ROOTDIR}/docs/index.html
+view-docs: docs
+	expect -c 'spawn PIPENV run mkdocs serve; expect "Serving on"; exec open "http://localhost:8000"; interact'
+
+.PHONY: publish-docs
+publish-docs:
+	if [ -z ${shell git status -uno | grep "Your branch is up to date with 'origin/main'"} ]; then \
+		echo "The documentation can only be published from main at head."; \
+		exit 1; \
+	fi
+	PIPENV run mkdocs gh-deploy
```

Comment:
This step is assuming that hosted docs live in gh-pages and can't be configured to respect an alternate directory?